packages/smart-wallet/src/smartWallet.js

```diff
@@ -366,6 +428,87 @@ export const prepareSmartWallet = (baggage, shared) => {
             },
           });
         },
+
+        provideBrandToPurses() {
+          const brandToPurses = provideLazy(
+            walletPurses,
+            this.facets.helper,
+            _k => {
+              /** @type {MapStore<Brand, PurseRecord[]>} */
+              const store = makeScalarBigMapStore('purses by brand', {
+                durable: true,
+              });
+              return store;
+            },
+          );
+          return brandToPurses;
+        },
+
+        /**
+         * Provide a purse given a NameHub of issuers and their
+         * brands.
+         *
+         * We current support only one NameHub, agoricNames, and
+         * hence one purse per brand. But we store an array of them
+         * to facilitate a transition decentralized introductions.
+         *
+         * @param {Brand} brand
+         * @param {ERef<NameHub>} known - namehub with brand, issuer branches
+         * @returns {Promise<Purse | undefined>} undefined if brand is not well known
+         */
+        async providePurseForKnownBrand(brand, known = shared.agoricNames) {
+          const brandToPurses = this.facets.helper.provideBrandToPurses();
+
+          if (brandToPurses.has(brand)) {
+            const purses = brandToPurses.get(brand);
+            if (purses.length > 0) {
+              // For now, we have just 1, so use it.
```

Comment:
```suggestion
              // UNTIL https://github.com/Agoric/agoric-sdk/issues/6126
```

Reply:
UNTIL doesn't seem to be documented like [XXX, TODO, etc.](https://github.com/Agoric/agoric-sdk/wiki/Coding-Style#code-comments). oversight?